src/Common/FileCache.cpp

```diff
@@ -151,9 +151,11 @@ FileSegments LRUFileCache::getImpl(
         return {};
     }
 
+    normalize(cells, cache_lock);
```

Comment:
not optimal enough (extra iteration) + memory in `files` will never be cleaned in case cache key is not accessed anymore